tests/system_tests_three_routers.py

```diff
@@ -220,6 +225,71 @@ def run(self):
         Container(self).run()
  
  
+class DynamicRequestResponseTest(MessagingHandler):
+    def __init__(self, address1, address2):
+        super(DynamicRequestResponseTest, self).__init__(prefetch=0)
+        self.address1   = address1
+        self.address2   = address2
+        self.dest       = "closest.whatever"
+        self.error      = None
+        self.sender     = None
+        self.server_receiver = None
+        self.client_receiver = None
+        self.n_expected = 10
+        self.n_sent     = 0
+        self.n_received = 0
+        self.n_accepted = 0
+ 
+    def timeout(self):
+        self.error = "Timeout Expired %d messages received." % self.n_received
+        self.conn1.close()
+        self.conn2.close()
+ 
+    def on_released(self, event):
+        self.n_sent -= 1
+        time.sleep(0.1)
+
+    def on_link_opened(self, event):
+        if event.receiver:
+            self.sender = event.container.create_sender(self.conn1, None)
```

Comment:
You are going to get two on_link_opened events for receivers.  This means you are creating the same sender twice.  This one sender is used for both client-send and server-send operations.